src/TcpAdapterProxy.cpp

```diff
@@ -171,6 +171,11 @@ namespace aws { namespace iot { namespace securedtunneling {
                 if (GET_SETTING(settings, WEB_SOCKET_DATA_ERROR_RETRY))
                 {
                     BOOST_LOG_SEV(log, error) << "Error from io_ctx::run(): " << e.what();
+                    if(e.is_local_port_bind_failure() && adapter_config.mode == proxy_mode::SOURCE) {
+                        BOOST_LOG_SEV(log, error) << "Local proxy failed to bind to port " << tac.adapter_config.data_port
+                            << " in source mode. Verify the selected port is not already in use and try again.";
+                        return;
```

Comment:
The result of this error should cause the local proxy process to return with an error code on the command line. That way, anyone scripting around the launching/running process may use a non-zero exit and know something abnormal happened. As it stands, returning normally here will cause main() to return normally even though the local proxy completely failed to run.

Reply:
Thanks for the feedback @DavidOgunsAWS, that's a great point that I hadn't considered. I have implemented your feedback and am now reporting a non-zero exit code to the command line in the event that the local proxy fails to bind to a local source port.